ocs_ci/deployment/factory.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import exceptions
+from .aws import AWSIPI
+
+logger = logging.getLogger(name=__file__)
+
+
+class DeploymentFactory(object):
+    """
+    A factory class to get specific platform object
+    """
+    def __init__(self):
+        # A map all existing deployments and respective classes
```

Comment:
Sorry I am late here, but trying to understand why we need a Factory for deployment?

Reply:
@vasukulkarni We want our cluster fixture to be minimal and ultimately not care about the implementation details of the deployment (i.e the provider). The factory is determining which type of deployment and returning the correct type of deployer to the cluster fixture which just calls the necessary methods from there.

Reply:
I think its a *overkill* here, since the init will know what to kind of deployment is initialized from the params and could be just one time based on CONFIG object we have,  Since I am late to this review here I am not expecting this to change now, but we should revisit this and fix.